api/app/auto_spatial_advisory/hfi_percent_conifer.py

```diff
@@ -0,0 +1,161 @@
+import argparse
+import asyncio
+import logging
+from datetime import date, datetime
+from time import perf_counter
+
+import numpy as np
+from geoalchemy2.shape import to_shape
+from osgeo import gdal, osr
+from sqlalchemy import select
+from sqlalchemy.ext.asyncio import AsyncSession
+from wps_shared import config
+from wps_shared.db.crud.auto_spatial_advisory import (
+    get_fire_zone_unit_shape_type_id,
+    get_fire_zone_units,
+    get_run_parameters_by_id,
+    get_run_parameters_id,
+    get_sfms_mixed_fuel_type,
+    get_table_srid,
+)
+from wps_shared.db.database import get_async_write_session_scope
+from wps_shared.db.models.auto_spatial_advisory import AdvisoryHFIPercentConifer, Shape
+from wps_shared.geospatial.geospatial import prepare_wkt_geom_for_gdal, rasters_match
+from wps_shared.logging import configure_logging
+from wps_shared.run_type import RunType
+from wps_shared.utils.s3 import set_s3_gdal_config
+from app.auto_spatial_advisory.common import get_hfi_s3_key
+
+osr.UseExceptions()
+gdal.UseExceptions()
+
+logger = logging.getLogger(__name__)
+
+
+def get_percent_conifer_s3_key(year: int):
```

Comment:
I'd make this a date that's passed in and have `get_percent_conifer_s3_key` be responsible for extracting the year, otherwise callers can call this and pass in any integer.